webapp/components/product-info.js

```diff
@@ -106,6 +106,16 @@ function parseProduct(name) {
   };
 }
 
+function createProduct(browserName, browserVersion, labels, revision) {
```

Comment:
This helper function should probably take a run instead. It'd make the call sites much simpler.
```suggestion
function productFromRun(run) {
```
You can simply `testRuns.map(productFromRun)`.